src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -122,17 +124,34 @@ private static Person createEditedPerson(Person personToEdit, EditPersonDescript
         Set<Tag> updatedTags = editPersonDescriptor.getTags().orElse(personToEdit.getTags());
         Optional<Integer> id = personToEdit.getId();
 
-        Person updatedPerson = new Person(updatedName, updatedPhone, updatedEmail, updatedAddress, updatedBirthday,
-                updatedLinkedin, updatedSecondaryEmail, updatedTelegram, updatedTags, id);
-
         if ((!personToEdit.hasValidBirthday() && !updatedBirthday.equals(Optional.empty()))
                 || (!personToEdit.hasValidLinkedin() && !updatedLinkedin.equals(Optional.empty()))
                 || (!personToEdit.hasValidTelegram() && !updatedTelegram.equals(Optional.empty()))
                 || (!personToEdit.hasValidSecondaryEmail() && !updatedSecondaryEmail.equals(Optional.empty()))) {
             throw new CommandException(MESSAGE_EDIT_ALTERNATIVE_FAIL);
-        } else if (updatedPerson.equals(personToEdit)) {
-            throw new CommandException(String.format(MESSAGE_EDIT_FIELDS_SAME, updatedName));
         } else {
+            if (personToEdit.hasValidBirthday() && updatedBirthday.equals(Optional.empty())) {
```

Comment:
Wondering if this can be implemented more succinctly. For instance, could we replace the lines above with

```java
Optional<Birthday> updatedBirthday = editPersonDescriptor.getBirthday().or(() -> personToEdit.getBirthday())
```

This way most of these lines below can be removed, other than the last few checks

Reply:
Yes, I absolutely agree! it is my mistake that I am looking at an older Java documentation for the `Optional` class so in a sense, I limited myself to the methods I can use. Thank you for pointing this out and suggesting a great alternative! Fixed in bc09b84